### ISSUE TITLE
Install `azure-cli` with `uv` fails

### ISSUE THREAD

**Issue by baggiponte** [astral-sh/uv#4685]
Referencing [this](https://github.com/astral-sh/uv/issues/2252#issuecomment-1985853109) comment #2252 but also #4600: some libraries require `wheel` to compile. How should we handle this when using `uv tool install`? (`rye` has the same issue.) Example:

```bash
uv tool install azure-cli
warning: `uv tool install` is experimental and may change without warning.
⠼ azure-mgmt-advisor==2.0.1                                                                                                                                                             error: Failed to download and build `jsmin==2.2.2`
  Caused by: Failed to build: `jsmin==2.2.2`
  Caused by: Build backend failed to determine extra requires with `build_wheel()` with exit status: 1
--- stdout:

--- stderr:
error in jsmin setup command: use_2to3 is invalid.
```

**Comment by baggiponte** [comment]
Oh, thanks. Now I get another error:

```bash
uv tool install azure-cli --prerelease=allow
...
error: Failed to find dist-info directory `azure-cli-2.61.0.dist-info` in environment at /Users/luca/Library/Application Support/uv/tools/azure-cli
```

Will rename the issue, I guess.

**Comment by baggiponte** [comment]
That was a speedy PR, wow. I wouldn't know how to rename the issue, feel free to suggest and/or close.

**Comment by konstin** [comment]
The title is correct :) I'll close it through the PR

**Comment by baggiponte** [comment]
> I which release version this would be fixed? 0.2.29 this still fails

that's odd. it works for me on the latest (and even previous ones) with `--prerelease=allow`.
